src/java.base/share/classes/java/lang/Object.java

```diff
@@ -52,7 +53,7 @@
  */
 @AnnotatedFor({"aliasing", "guieffect", "index", "lock", "nullness"})
 @PolyUIType
-public class Object {
+public @UsesObjectEquals class Object {
```

Comment:
The [Javadoc for @UsesObjectEquals](https://checkerframework.org/api/org/checkerframework/checker/interning/qual/UsesObjectEquals.html) says:

> A class may be annotated @UsesObjectEquals if neither it, nor any of its supertypes or subtypes, overrides equals. Therefore, it cannot be written on Object itself.